app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -445,41 +442,41 @@ public void downloadData(DataDownloadEvent event) {
         Timber.d("Download has started");
     }
 
-    @Subscribe
-    public void ErrorHandlerEvent(RetrofitError cause) {
-        ConnectivityManager connMgr = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
-        NetworkInfo netinfo = connMgr.getActiveNetworkInfo();
-        if (!(netinfo != null && netinfo.isConnected())) {
-            OpenEventApp.postEventOnUIThread(new ShowNetworkDialogEvent());
-        } else {
-            switch (cause.getKind()) {
-                case CONVERSION: {
-                    errorType = "Conversion Error";
-                    errorDesc = String.valueOf(cause.getCause());
-                    break;
-                }
-                case HTTP: {
-                    errorType = "HTTP Error";
-                    errorDesc = String.valueOf(cause.getResponse().getStatus());
-                    break;
-                }
-                case UNEXPECTED: {
-                    errorType = "Unexpected Error";
-                    errorDesc = String.valueOf(cause.getCause());
-                    break;
-                }
-                case NETWORK: {
-                    errorType = "Network Error";
-                    errorDesc = String.valueOf(cause.getCause());
-                    break;
-                }
-                default: {
-                    errorType = "Other Error";
-                    errorDesc = String.valueOf(cause.getCause());
-                }
-            }
-            Timber.tag(errorType).e(errorDesc);
-            showErrorDialog(errorType, errorDesc);
-        }
-    }
+//    @Subscribe
```

Comment:
this method was not used, but I see it could be a good way of handling errors but it's not compatible with retrofit2 so I commented it out so I can reuse the code later or anyone who interested in creating an effective error handling method.